### PR TITLE
Add @hashicorp/platform-configs package

### PR DESCRIPTION
**Pull request by dstaley**
🎟️ [Asana Task](https://app.asana.com/0/0/1202025665826438/f)

---

## Description

Adds a new `@hashicorp/platform-configs` package to share basic config files that don't have any dependencies.

## PR Checklist 🚀

- [ ] Conduct thorough self-review.
- [ ] Add or update tests as appropriate.
- [ ] Write a useful description (above) to give reviewers appropriate context.
- [ ] Identify (in the description above) and document (add Asana tasks on [this board](https://app.asana.com/0/1100423001970639/list)) any technical debt that you're aware of, but are not addressing as part of this PR.

### CODE REVIEW

File: packages/configs/tsconfig/next.json [hunk 1]
```diff
@@ -0,0 +1,22 @@
+{
+  "$schema": "https://json.schemastore.org/tsconfig",
```
Comment (BRKalow) [inline packages/configs/tsconfig/next.json:2]: ✨

File: packages/configs/README.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
+# `@hashicorp/platform-configs`
+
+This package provides shared configuration files that can be used in HashiCorp projects.
+
+## Available Config Files
+
+### `tsconfig/next.json`
+
+Add to your `tsconfig.json`:
```
Comment (BRKalow) [inline packages/configs/README.md:9]: Why is this one called `next.json`? It would be helpful to specify here!

**Review by BRKalow — APPROVED** [review]
Looks good! Are you thinking we'll rip the rest out of `platform-cli` at a later date?

**Comment by dstaley** [comment]
> Are you thinking we'll rip the rest out of `platform-cli` at a later date?

Yup, I didn't know how well `changesets` would react to me pushing up an update that references an unpublished package, so we can adjust that one in a follow-up PR.

**Comment by dstaley** [comment]
This is working well in canary on dev portal, so I'mma merge this :tada:

**Pull request merged**